### PR TITLE
#927 route starting and stopping of bluetooth discovery through new functions startDiscovery() and stopDiscovery()

### PR DESCRIPTION
**Pull request by drmason789**
#927

### CODE REVIEW

File: src/bluetooth.cpp [hunk 1]
```diff
@@ -172,16 +155,34 @@ void bluetooth::finished() {
...
+                                  QBluetoothDeviceDiscoveryAgent::LowEnergyMethod);
+        }
+    #endif
+}
+
+
+void bluetooth::stopDiscovery()
+{
+    if(this->discoveryAgent)
+        this->discoveryAgent->stop();
```
Comment (drmason789) [inline src/bluetooth.cpp:185]: This is changed behaviour...previous code was not doing the nullptr check.

Comment (cagnulein) [reply src/bluetooth.cpp:185]: let's put a qdebug if the discoveryagent is a null pointer in order to track future bug, thanks

File: src/bluetooth.cpp [hunk 2]
```diff
@@ -172,16 +155,34 @@ void bluetooth::finished() {
...
+}
+
+void bluetooth::startDiscovery() {
+
+    #ifndef Q_OS_IOS
+        QSettings settings;
+        bool technogym_myrun_treadmill_experimental =
+            settings.value(QStringLiteral("technogym_myrun_treadmill_experimental"), false).toBool();
+        bool trx_route_key = settings.value(QStringLiteral("trx_route_key"), false).toBool();
+        bool bh_spada_2 = settings.value(QStringLiteral("bh_spada_2"), false).toBool();
```
Comment (drmason789) [inline src/bluetooth.cpp:168]: I moved these bools into the #ifndef block because they're not used outside of it.

Comment (cagnulein) [reply src/bluetooth.cpp:168]: sounds good

File: src/bluetooth.cpp (hunk 1 above)
Comment (drmason789) [reply src/bluetooth.cpp:185]: Would you prefer one that always happens, or all call to debug(..) which checks the log field ?

Comment (cagnulein) [reply src/bluetooth.cpp:185]: In the else case as you did, thanks!

**Pull request merged**